release/models/ipsec/openconfig-ipsec-offload.yang

```diff
@@ -0,0 +1,357 @@
+    }
+    
+    leaf ext-seq-num {
+      type boolean;
+      default "true";
+      description
+        "True if this IPsec SA is using extended
+        sequence numbers.  If true, the 64-bit
+        extended sequence number counter is used;
+        if false, the normal 32-bit sequence
+        number counter is used.";
+    }
+
+    leaf anti-replay-window-size {
+      type uint32;
+      default "64";
+      description
+        "To set the anti-replay window size.
+        The default value is set to 64,
+        following the recommendation in RFC 4303.";
+      reference
+        "RFC 4303: IP Encapsulating Security Payload (ESP),
+        Section 3.4.3.";
+    }
+
+    leaf protocol-parameters {
+      type ietf-nsfikec:ipsec-protocol-params;
+      default "esp";
+      description
+        "Security protocol of IPsec SA, only
+        ESP so far.";
+    }
+    
+    leaf mode {
+      type ietf-nsfikec:ipsec-mode;
+      default "transport";
+      description
+        "Tunnel or transport mode.";
+    }
+    
+    container esp-sa {
+      description
+        "In case the IPsec SA is an
+        Encapsulation Security Payload
+        (ESP), it is required to specify
+        encryption and integrity
+        algorithms and key materials.";
+      
+      container encryption {
+        description
+          "Configuration of encryption or
+          AEAD algorithm for IPsec
+          Encapsulation Security Payload
+          (ESP).";
+        
+        leaf encryption-algorithm {
+          type ietf-nsfikec:encr-alg-t;
+          default "12";
+          description
+            "Configuration of ESP
+            encryption.  With AEAD
+            algorithms, the integrity-algorithm
+            leaf is not used.";
+        }
+        
+        leaf key {
+          ietf-nacm:default-deny-all;
+          type ietf-yang:hex-string;
+          mandatory true;
+          description
+            "ESP encryption key value.
+            If this leaf is not defined,
+            the key is not defined
+            (e.g., encryption is NULL).
+            The key length is
+            determined by the
+            length of the key set in
+            this leaf.  By default, it is
+            128 bits.";
+        }
+        
+        leaf key-len {
+          type uint32;
+          mandatory true;
+          description
+            "ESP encryption key length value.
+            The key in this leaf node contains
+            the salt as well. The len field will
+            determine the length and remaining
+            will be used as salt.";
+        }
+
+      }
+
+    } /*container esp-sa*/
+    
+    container sa-lifetime-hard {
+      description
+        "IPsec SA hard lifetime.  The action
+        associated is terminate and hold.";
+      uses ietf-nsfikec:lifetime;
+    }
+    
+    container sa-lifetime-soft {
+      description
+        "IPsec SA soft lifetime.";
+      uses ietf-nsfikec:lifetime;
```

Comment:
Does this require include of a file?

Reply:
If you are running a `pyang` (or other lint) command to compile and validate the tree, then yes. The files have to be included in the path. But that is outside the context of this file and looking at other OpenConfig models, they expect the user to find them and point to the right include directory.

However, this one is a bit different from that model, because `ietf-nsfikec` is not part of the OpenConfig release -- it has to be downloaded from another repo (github.com/YangModels). Leaving it out for now for initial version